File: 035/cpp/main.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int i = -1;
        int j = nums.size();
        while (i + 1 < j) {
            int k = i + (j - i) / 2;
            if (nums[k] < target) {
                i = k;
            } else {
                j = k;
            }
        }
        return i + 1;
    }
};

File: 035/rust/main.rs
impl Solution {
    pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
        let mut i = -1;
        let mut j = nums.len() as i32;
        while i + 1 < j {
            let k = (i + (j - i) / 2) as usize;
            if nums[k] < target {
                i = k as i32;
            } else {
                j = k as i32;
            }
        }
        return i + 1;
    }
}
